dipy/denoise/localpca.py

```diff
@@ -66,6 +94,99 @@ def _pca_classifier(L, nvoxels):
     return var, ncomps
 
 
+def create_patch_radius_arr(arr, pr):
+    """Create the patch radius array from the data to be denoised and the patch
+    radius.
+
+    Parameters
+    ----------
+    arr : ndarray
+        Data to be denoised.
+    pr : int or ndarray
+        Patch radius.
+
+    Returns
+    -------
+    patch_radius : ndarray
+        Number of samples.
+    """
+
+    patch_radius = copy.deepcopy(pr)
+
+    if isinstance(patch_radius, int):
+        patch_radius = np.ones(3, dtype=int) * patch_radius
+    if len(patch_radius) != 3:
+        raise ValueError("patch_radius should have length 3")
+    else:
+        patch_radius = np.asarray(patch_radius).astype(int)
+    patch_radius[arr.shape[0:3] == np.ones(3)] = 0  # account for dim of size 1
+
+    return patch_radius
+
+
+def compute_patch_size(patch_radius):
+    """Compute the patch size from the patch radius: it is twice the radius plus
+    one.
+
+    Parameters
+    ----------
+    patch_radius : ndarray
+        Patch radius.
+
+    Returns
+    -------
+    int
+        Number of samples.
+    """
+
+    return 2 * patch_radius + 1
+
+
+def compute_num_samples(patch_size):
+    """Compute the number of samples as the dot product of the elements.
+
+    Parameters
+    ----------
+    patch_size : ndarray
+        Patch size.
+
+    Returns
+    -------
+    int
+        Number of samples.
+    """
+
+    return np.prod(patch_size)
+
+
+def compute_suggested_patch_radius(arr, patch_size):
```

Comment:
This could have taken `arr.shape[-1]` as the first argument, but I thought it was less clear from the point of view of the caller.